src/Config/DatabasePrimitives/DatabaseObject.cs

```diff
@@ -42,7 +42,9 @@ public bool Equals(DatabaseObject? other)
     {
         return other is not null &&
                SchemaName.Equals(other.SchemaName) &&
-               Name.Equals(other.Name);
+               Name.Equals(other.Name) &&
+               FullName.Equals(other.FullName) &&
+               ((SourceDefinition == null && other.SourceDefinition == null) || SourceDefinition?.Equals(other.SourceDefinition) == true);
```

Comment:
if source definition is null and other.sourcedefinition is null, wont this condition evaluate to true? Is that the expectation.

Reply:
yes that is the expectation, both the object should be equal if they are null or not.

Reply:
I meant is not null sorry. Like one is null and the other is not.

Reply:
if one is null and other is not then second half of the condition will return false.

SourceDefinition?.Equals(other.SourceDefinition) == true
scenario 1 : SourceDefinition is null and other.SourceDefinition is not null
the above will translate to null == true which will return false

scenario 2: Source Definition is not null and other.SourceDefinition is null
SourceDefinition.Equals(null) will return false.

Reply:
tested with a unit test and sent screenshot offline